### ISSUE TITLE
Constrained TypeVars with Any default

### ISSUE THREAD

**Issue by srittau** [python/mypy#20184]
**Bug Report**

In python/typeshed#13683 I'm trying to change a type var to the following:

```python
_I = TypeVar("_I", int, bytes, float, bool, default=Any)  # possibly memoryview types
```

Unfortunately, mypy complains: `error: TypeVar default must be one of the constraint types  [misc]`. While this is of course generally sound advice, a default of `Any` should always be permissible, as this allows `memoryview` to stand for any possible memoryview type. Without it, users are forced to write `memoryview[Any]` in every instance, which is unreasonable.

**Your Environment**

- Mypy version used: 1.18.2

**Comment by sterliakov** [comment]
I agree with your position, yet [the spec](https://typing.python.org/en/latest/spec/generics.html#constraints) seems to disallow that:

> For constrained TypeVars, the default needs to be one of the constraints. A type checker should generate an error even if it is a subtype of one of the constraints.

**Comment by srittau** [comment]
Thanks, for pointing this out. I've opened a discussion at https://discuss.python.org/t/constrained-type-vars-with-any-default/104799.
